### PR TITLE
add INTELLLVM support.

### PR DESCRIPTION
**Pull request by robinson96**
Teach BLT about the Intel LLVM OneAPI compiler.

### CODE REVIEW

File: cmake/BLTMacros.cmake [hunk 1]
```diff
@@ -1017,9 +1018,12 @@ macro(blt_append_custom_compiler_flag)
        elseif( DEFINED arg_PGI AND Fortran_COMPILER_FAMILY_IS_PGI )
           set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_PGI} " )
        elseif( DEFINED arg_GNU AND Fortran_COMPILER_FAMILY_IS_GNU )
+          message("appending ${arg_GNU} to ${arg_FLAGS_VAR}")
```
Comment (adayton1) [inline cmake/BLTMacros.cmake:1021]: Should this message be removed?

File: cmake/SetupCompilerOptions.cmake [hunk 2]
```diff
@@ -70,6 +74,10 @@ else()
         set(Fortran_COMPILER_FAMILY_IS_INTEL 1)
         message(STATUS "Fortran Compiler family is Intel")
 
+    elseif("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "IntelLLVM")
+        set(Fortran_COMPILER_FAMILY_IS_INTELLLVM 1)
+        message(STATUS "Fortran Compiler family is Intel")
```
Comment (adayton1) [inline cmake/SetupCompilerOptions.cmake:79]: IntelLLVM

**Review by kennyweiss — CHANGES_REQUESTED** [review]
Thanks @robinson96 

* Please add a line to the `RELEASE_NOTES`
* Also, does this need to be incorporated into existing usages of `blt_append_custom_compiler_flag`?
https://github.com/LLNL/blt/blob/91ba4aa22b5bcdc792d644f17c25efc94421d9fd/cmake/SetupCompilerOptions.cmake#L300-L351

**Comment by robinson96** [comment]
> 
> 
> Thanks @robinson96
> 
>     * Please add a line to the `RELEASE_NOTES`
> 
>     * Also, does this need to be incorporated into existing usages of `blt_append_custom_compiler_flag`?
>       https://github.com/LLNL/blt/blob/91ba4aa22b5bcdc792d644f17c25efc94421d9fd/cmake/SetupCompilerOptions.cmake#L300-L351

Release note added. The default values for all of those calls look to be aligned with what is necessary for IntelLLVM.

**Review by adayton1 — APPROVED** [review]
LGTM

**Comment by adayton1** [comment]
@kennyweiss, @white238, any more comments on this pull request, or can it be merged?

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
@kennyweiss can you rereview then dismiss your change request?  After that I will create a local branch from the fork branch, it runs through the gitlab tests.

**Review by kennyweiss — APPROVED** [review]
Thanks @adayton1 -- I just noticed that the API docs will also need to be updated to reflect the new addition:
https://github.com/LLNL/blt/blob/develop/docs/api/utility.rst

Other than that, this should be good to go.

**Comment by adayton1** [comment]
This is on Peter's personal fork, and he's out today. If you want to just add it to the branch you'll create from his fork, that's fine. Otherwise, I'll ask him to fix the API docs when he gets back in on Monday.

**Pull request merged**